### PR TITLE
Add README for IceRpc.Protobuf.Tools

### PR DESCRIPTION
**Pull request by pepone**
Added the IceRpc.Protobuf.Tools README, this is included with the corresponding NuGet package

Renamed `ProtoImportPath` to `ProtoDirectory` for consistency with IceRpc.Slice.Tools.

### CODE REVIEW

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+generator is a .NET program and requires .NET 8 or later.
+
+Once you've added the IceRpc.Protobuf.Tools NuGet package to your project, the Protobuf files of your project are
+automatically compiled into C# files every time you build this project.
+
+[Source code][source] | [Package][package]
+
+## Protobuf files and Protobuf directories
+
+The Protobuf compiler accepts two main inputs:
```
Comment (externl) [inline tools/IceRpc.Protobuf.Tools/README.md:17]: We're referring to `protoc` here right?

Comment (pepone) [reply tools/IceRpc.Protobuf.Tools/README.md:17]: Yes

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 2]
```diff
@@ -1,2 +1,79 @@
+# Protobuf Tools for IceRPC
 
-TODO
+IceRpc.Protobuf.Tools allows you to compile Protobuf definitions (in `.proto` files) into C# code (in `.cs` files)
+within MSBuild projects.
+
+This package includes the Protobuf compiler, `protoc`, and the `protoc-gen-icerpc-csharp` generator. The `protoc`
```
[the remark this answers is not in the document]
Comment (pepone) [reply tools/IceRpc.Protobuf.Tools/README.md:6]: It includes `3.24.4`, we can add the version here if you feel it is important.

[the remark this answers is not in the document]
Comment (bernardnormier) [reply tools/IceRpc.Protobuf.Tools/README.md:6]: I wouldn't include the version.

[the remark this answers is not in the document]
Comment (bernardnormier) [reply tools/IceRpc.Protobuf.Tools/README.md:6]: It would be nice to clarify that protoc (as I understand it) incorporates a number of (code) generators, including one for C#. So plain protoc can generate C# from .proto definition, and we need the additional protoc-gen-icerpc-csharp, included in this package, to generate IceRPC + Protobuf integration code for C#.

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 3]
```diff
@@ -1,2 +1,79 @@
...
+compiler is a native tool with binaries for Linux x64, macOS x64 and Windows (x64). The `protoc-gen-icerpc-csharp`
```
Comment (bernardnormier) [inline tools/IceRpc.Protobuf.Tools/README.md:7]: Should be consistent wrt parenthesis.

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 4]
```diff
@@ -1,2 +1,79 @@
...
+generator is a .NET program and requires .NET 8 or later.
+
+Once you've added the IceRpc.Protobuf.Tools NuGet package to your project, the Protobuf files of your project are
+automatically compiled into C# files every time you build this project.
```
Comment (bernardnormier) [inline tools/IceRpc.Protobuf.Tools/README.md:11]: Maybe that's where we should mention the plain .cs and the .icerpc.cs files?

File: examples/GreeterProtobuf/Client/Client.csproj [hunk 5]
```diff
@@ -10,7 +10,7 @@
   </PropertyGroup>
   <ItemGroup>
     <ProtoFile Include="../proto/greeter.proto">
-      <ProtoImportPath>../proto</ProtoImportPath>
+      <ProtoDirectory>../proto</ProtoDirectory>
```
Comment (bernardnormier) [inline examples/GreeterProtobuf/Client/Client.csproj:13]: I would revert this change: I find ProtoImportPath more correct.

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 6]
```diff
@@ -1,2 +1,79 @@
...
+
+[Source code][source] | [Package][package]
+
+## Protobuf files and Protobuf directories
+
+The Protobuf compiler accepts two main inputs:
+
+- the Protobuf files to compile into C# code (the Protobuf files)
+- directories that contain reference Protobuf files (the Protobuf directories)
```
Comment (bernardnormier) [inline tools/IceRpc.Protobuf.Tools/README.md:20]: I find this misleading. I'd rather call it the "import path".

A big difference between protoc and slicec is protoc is like a C++ compiler.
slicec compiles all the .slice files it finds in the "directories". protoc does not compile all the .proto files in its import path. The import path is about locating imported files and protoc compiles only source files and the files these source files import.

File: tools/IceRpc.Protobuf.Tools/README.md [hunk 7]
```diff
@@ -1,2 +1,79 @@
...
+By default, all `.proto` files located in your project's home directory and any of its subdirectories, recursively, are
+included in `ProtoFile`. You can prevent this auto-inclusion of `.proto` files by setting either
+[`EnableDefaultItems`][default-items] or `EnableDefaultProtoFileItems` to `false`. The default value of these properties
+is `true`.
+
+You can also add Protobuf files to your project explicitly. For example:
+
+```xml
+<ItemGroup>
+    <ProtoFile Include="../Greeter.proto"/>
```
Comment (bernardnormier) [inline tools/IceRpc.Protobuf.Tools/README.md:34]: We must use the standard naming conventions for .proto files. So greeter.proto here.

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]
